vyper/ast/grammar.lark

```diff
@@ -33,14 +33,19 @@ import: _IMPORT DOT* _import_path [import_alias]
       | _import_from _IMPORT ( WILDCARD | _import_name [import_alias] )
       | _import_from _IMPORT "(" import_list ")"
 
-
 // Constant definitions
 // NOTE: Temporary until decorators used
-constant_def: NAME ":" "constant" "(" type ")" "=" _expr
+constant: "constant" "(" type ")"
+constant_private: NAME ":" constant
+constant_with_getter: NAME ":" "public" "(" constant ")"
+constant_def: (constant_private | constant_with_getter) "=" _expr
 
 // immutable definitions
 // NOTE: Temporary until decorators used
-immutable_def: NAME ":" "immutable" "(" type ")"
+immutable: "immutable" "(" type ")"
+immutable_private: NAME ":" immutable
+immutable_with_getter: NAME ":" "public" "(" immutable ")"
+immutable_def: immutable_private | immutable_with_getter
 
```

Comment:
can we refactor this section a little? probably the definition of `variable` below can be changed from
```
variable_with_getter: NAME ":" "public" "(" type ")"
```
to 
```
variable_with_getter: NAME ":" "public" "(" type | immutable | constant ")"
```

Reply:
yes but there would be an issue because you can't assign values to storage variables on declaration but you have to assign one to a constant.

so I could do: 
`variable_with_getter: NAME ":" "public" "(" type | immutable ")"`
but I would still need a separate `constant_with_getter` because we can't have:
`constant_def: (constant_private | variable_with_getter) "=" _expr`

so it would be:

```
// Constant definitions
// NOTE: Temporary until decorators used
constant: "constant" "(" type ")"
constant_private: NAME ":" constant
constant_with_getter: NAME ":" "public" "(" constant ")"
constant_def: (constant_private | constant_with_getter) "=" _expr

// immutable definitions
// NOTE: Temporary until decorators used
immutable: "immutable" "(" type ")"
immutable_private: NAME ":" immutable

variable: NAME ":" type
// NOTE: Temporary until decorators used
variable_with_getter: NAME ":" "public" "(" type | immutable ")"
variable_def: variable | variable_with_getter
```